src/sqlfluff/core/parser/parsers.py

```diff
@@ -123,15 +123,17 @@ def __init__(
             **segment_kwargs,
         )
 
-    def simple(cls, parse_context: ParseContext, crumbs=None):
+    def simple(
+        cls, parse_context: ParseContext, crumbs=None
+    ) -> Tuple[frozenset, frozenset]:
```

Comment:
To keep things consistent around this more convoluted type, I recently added `SimpleHintType` to characterise the return values of `simple` types. It's currently in `grammar/base.py`, although for this implementation that definition should probably move somewhere more central. I think it's a complicated enough type that we should use that single defined type for consistency.

I won't comment all the other like this - but there are a few.